worker/caasoperatorprovisioner/worker.go

```diff
@@ -114,35 +142,51 @@ func (p *provisioner) loop() error {
 				if err != nil {
 					return errors.Trace(err)
 				}
-				appPasswords = append(appPasswords, apicaasprovisioner.ApplicationPassword{Name: app, Password: password})
+				newApps = append(newApps, apicaasprovisioner.ApplicationPassword{Name: app, Password: password})
 			}
-			if len(appPasswords) == 0 {
+			if len(newApps) == 0 {
 				continue
 			}
-			errorResults, err := p.provisionerFacade.SetPasswords(appPasswords)
-			if err != nil {
-				return errors.Annotate(err, "failed to set application api passwords")
+			if err := p.ensureOperators(newApps); err != nil {
+				return errors.Trace(err)
 			}
-			var errorStrings []string
-			for i, r := range errorResults.Results {
-				if r.Error != nil {
-					errorStrings = append(errorStrings, r.Error.Error())
-					continue
-				}
-				if err := p.createOperator(appPasswords[i].Name, appPasswords[i].Password); err != nil {
-					return errors.Trace(err)
-				}
-			}
-			if errorStrings != nil {
-				err := errors.New(strings.Join(errorStrings, "\n"))
-				return errors.Annotate(err, "failed to set application api passwords")
+			// Store the apps we have just added.
+			for _, ap := range newApps {
+				p.appPasswords[ap.Name] = ap.Password
 			}
+
+			// Now we have been through all the applications at least once, we can
+			// listen for api address changes.
+			apiAddressChanged = apiAddressWatcher.Changes()
 		}
 	}
 }
 
-func (p *provisioner) createOperator(app, password string) error {
-	logger.Debugf("creating operator for %q", app)
+// ensureOperators creates operator pods for the specified app names -> api passwords.
+func (p *provisioner) ensureOperators(appPasswords []apicaasprovisioner.ApplicationPassword) error {
+	errorResults, err := p.provisionerFacade.SetPasswords(appPasswords)
+	if err != nil {
+		return errors.Annotate(err, "failed to set application api passwords")
+	}
+	var errorStrings []string
+	for i, r := range errorResults.Results {
+		if r.Error != nil {
+			errorStrings = append(errorStrings, r.Error.Error())
+			continue
+		}
+		if err := p.ensureOperator(appPasswords[i].Name, appPasswords[i].Password); err != nil {
+			return errors.Trace(err)
+		}
+	}
+	if errorStrings != nil {
+		err := errors.New(strings.Join(errorStrings, "\n"))
+		return errors.Annotate(err, "failed to set application api passwords")
+	}
+	return nil
+}
+
+func (p *provisioner) ensureOperator(app, password string) error {
+	logger.Debugf("ensuring operator for %q", app)
```

Comment:
trace?

Reply:
Removed the debug, instead:
In machine provisioner we log an Info message when machine is started successfully.
I've added an Info message for when the operator has started as its a similar operation and useful to know.